components/board/server.go

```diff
@@ -167,6 +167,24 @@ func (s *serviceServer) ReadAnalogReader(
 	return &pb.ReadAnalogReaderResponse{Value: int32(val)}, nil
 }
 
+// WriteAnalog writes the analog value to the analog writer pin of the underlying robot.
+func (s *serviceServer) WriteAnalog(
+	ctx context.Context,
+	req *pb.WriteAnalogRequest,
+) (*pb.WriteAnalogResponse, error) {
+	b, err := s.coll.Resource(req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	err = b.WriteAnalog(ctx, req.Pin, req.Value, req.Extra.AsMap())
```

Comment:
I think this implementation is cleaner and removes the need for both analogwriter helper functions. but the one downside is that we can't verify serverside that the request has a valid analog writer pin. The individual drivers would need to handle that.

Reply:
We rely on s.coll.Resource to give us an error in every other API, and the driver to return an error in every other component's server. So I think that might be overkill?